docs/Blog-Post.md

```diff
@@ -0,0 +1,578 @@
+We were initially excited about `xmap` when we first encountered it, but 1) they seem to be deprioritizing it (in favor of `pjit`)
+and 2) ultimately `xmap` can be confusing because you write non-named code for positional axes, then add names "outside"
+of the main model code itself. Ultimately, it's harder reason about this mixed positional/named code (where the axis names are implicit)
+than just using names the whole way through.
+
+Flax supports a logical-to-physical axis mapping thing similar to what's in Haliax. However, the arrays don't carry around
+their axis names, so you have to remember them and pass them in manually when doing partitioning for data parallelism,
+tensor parallism and FSDP. To us, this seems like a bit of a missed opportunity (relative to what we have in Haliax), but it's
+still useful.
+
+Haliax's NamedArrays are probably most similar to [Mesh-Tensorflow](https://github.com/tensorflow/mesh). PyTorch has
+[Named Tensors](https://pytorch.org/docs/stable/named_tensor.html). They're fairly new and "bolted on" to the existing
+positional tensors. They also don't help with model partitioning, which is one of the main use cases for named axes in
+Haliax.
+
+## Haliax Tutorials
+
+This is just a taste of what Haliax can do. For more details, please see our interactive tutorials on Colab:
+
+* [Introduction to Haliax with Transformers](https://colab.research.google.com/drive/1TiTcQQ4V5mopbgCu1SVl-oqJtXn7rFnC?usp=sharing)
+* [Scaling Transformers in Haliax](https://colab.research.google.com/drive/1QX4yH3zRFF3Xiibf1aahETcSQ5nbcUMz?usp=sharing), including FSDP in JAX.
+
+# Levanter: Bitwise Reproducible Foundation Models with JAX
+
+Levanter is a library for training foundation models built on top of Haliax. It provides a complete pipeline
+for training a GPT-2-like Transformer, complete with data preparation, logging, training, checkpointing, evaluation, and
+export, while maintaining bitwise reproducibility throughout.
+
+We have used Levanter to train models as large as 6.7b parameters on a v3-256, and have run experiments showing that it
+can scale up to least 65b parameters.
+
+## Bitwise Reproducibility
+
+One of the benefits of JAX is that it offers strong guarantees for reproducibility. In particular, JAX's fine-grained
+control over PRNG states makes it easy to ensure bitwise reproducibility, especially when using TPUs.
+Levanter takes advantage of this to offer bitwise reproducibility for training runs, even after preemption. In particular,
+the same run with the same code on the same set of hardware (e.g. a v3-32 or a v3-256) will produce the exact same loss curve, even if it is
+preempted and resumed multiple times. As an example, here is a screenshot of a training run being resumed multiple times, even on different TPU pod slices:
+
+![plot showing bitwise reproducibility with four training runs superimposed with the exact same loss curve](figures/bitwise_repro_curve.png)
```

Comment:
Can we show two Pytorch loss curves which are different but are the same code with the same random seed?  I don't think people might appreciate this difference.

Also, could we argue more passionately why reproducibility is nice.  For example, you get loss spikes, but they go away when you run it...it's like a Heissenbug or whack-a-hole - whatever analogy you want to use.

Reply:
i'll try to get a thing